### PR TITLE
add deck functionality

### PR DESCRIPTION
**Pull request by sonyashbo**
Add deck functionality implemented

### CODE REVIEW

File: src/app/decks/decks-list/decks-list.component.html [hunk 1]
```diff
@@ -5,4 +5,36 @@
      [routerLink]="['/deck/', set.id]">
     {{ set.name }}
   </a>
+  <button class="btn btn-success m-1" *ngIf="!editMode" (click)="onEditList()">
+    <i class="fas fa-plus"></i> Deck
+  </button>
+
+
+<form (ngSubmit)="onSubmit(f)" #f="ngForm">
```
Comment (albanul) [inline src/app/decks/decks-list/decks-list.component.html:13]: Please put all angular directives at the end of the tag

File: src/app/decks/decks-list/decks-list.component.html [hunk 2]
```diff
@@ -5,4 +5,36 @@
...
+    <div class="col-14 form-group">
```
Comment (albanul) [inline src/app/decks/decks-list/decks-list.component.html:14]: Consider to keep your html tags tight as possible

File: src/app/decks/decks-list/decks-list.component.html [hunk 3]
```diff
@@ -5,4 +5,36 @@
      [routerLink]="['/deck/', set.id]">
     {{ set.name }}
   </a>
+  <button class="btn btn-success m-1" *ngIf="!editMode" (click)="onEditList()">
+    <i class="fas fa-plus"></i> Deck
+  </button>
+
+
```
[the remark this answers is not in the document]
Comment (sonyashbo) [reply src/app/decks/decks-list/decks-list.component.html:12]: Done

File: src/app/decks/decks-list/decks-list.component.ts [hunk 4]
```diff
@@ -1,20 +1,38 @@
...
   ngOnInit(): void {
     this.setsState = this.setService.state;
   }
+
+  onEditList(): void {
+    this.editMode = true;
+  }
+
+  onCancel(): void {
+    this.newDeckName = '';
```
[the remark this answers is not in the document]
Comment (sonyashbo) [reply src/app/decks/decks-list/decks-list.component.ts]: Done

File: src/app/decks/decks-list/decks-list.component.ts [hunk 5]
```diff
@@ -1,20 +1,38 @@
...
+    this.editMode = false;
+  }
+
+  onSubmit(f: NgForm): void {
+    this.setService.addDeck(new Deck(-1, f.value.deckName));
```
[the remark this answers is not in the document]
Comment (sonyashbo) [reply src/app/decks/decks-list/decks-list.component.ts]: Done

File: src/app/decks/decks-list/decks-list.component.ts [hunk 6]
```diff
@@ -1,20 +1,38 @@
...
   selector: 'app-sets-list',
-  templateUrl: './decks-list.component.html'
+  templateUrl: './decks-list.component.html',
+  styleUrls: ['./decks-list.component.less']
 })
 export class DecksListComponent implements OnInit {
   setsState: Observable<DeckState>;
-
+  editMode: boolean;
+  newDeckName: string;
```
[the remark this answers is not in the document]
Comment (sonyashbo) [reply src/app/decks/decks-list/decks-list.component.ts]: Done

File: src/app/decks/decks-list/decks-list.component.html (hunk 1 above)
Comment (sonyashbo) [reply src/app/decks/decks-list/decks-list.component.html]: Done

File: src/app/decks/decks-list/decks-list.component.html (hunk 2 above)
Comment (sonyashbo) [reply src/app/decks/decks-list/decks-list.component.html:14]: Done

**Pull request merged**